### PR TITLE
reducesql: also attempt to remove columns from PKs

### PR DESCRIPTION
**Pull request by madelynnblue**
Release note: None

### CODE REVIEW

File: pkg/testutils/reduce/reducesql/reducesql.go [hunk 1]
```diff
@@ -672,14 +672,19 @@ var (
...
+			n := len(idx.Columns)
+			if xfi < len(idx.Columns) {
+				idx.Columns = append(idx.Columns[:xfi], idx.Columns[xfi+1:]...)
 			}
 			return n
 		}
+		switch node := node.(type) {
+		case *tree.IndexTableDef:
+			return removeCol(node)
+		case *tree.UniqueConstraintTableDef:
```
Comment (rohany) [inline pkg/testutils/reduce/reducesql/reducesql.go]: do you want to case on whether or not `PrimaryKey` is true, or just apply it to all `UniqueConstraintTableDef`s?

File: pkg/testutils/reduce/reducesql/reducesql.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ var SQLPasses = []reduce.Pass{
 	replaceStmt,
 	removeWithCTEs,
 	removeWith,
+	removeCreateDefs,
```
Comment (rohany) [inline pkg/testutils/reduce/reducesql/reducesql.go]: For my information, does the order of these matter?

**Review by rohany — APPROVED** [review]

File: pkg/testutils/reduce/reducesql/reducesql.go (hunk 2 above)
Comment (mjibson) [reply pkg/testutils/reduce/reducesql/reducesql.go]: The order matters. It's the order in which removals are attempted. So things that have larger impact should be executed earlier which will hopefully decrease the size of the statement as fast as possible, thus decreasing the runtime of the algorithm.

File: pkg/testutils/reduce/reducesql/reducesql.go (hunk 1 above)
Comment (mjibson) [reply pkg/testutils/reduce/reducesql/reducesql.go]: All index types benefit from this reduction, so yes, it should apply to both here.

**Pull request merged**